src/fun/counting/main.go.tmpl

```diff
@@ -0,0 +1,89 @@
+{{ $RoleID := false }} {{/* Counting role ID; set to false to disable */}}
+{{ $SecondChance := true }} {{/* Second chance if wrong; true/false */}}
+{{ $Leaderboard := true }} {{/* If you added the leaderboard CC; true/false */}}
+{{ $CorrectEmoji := "✅" }} {{/* Emoji to react with if number is correct; if custom, use format name:id */}}
+{{ $WarningEmoji := "⚠️" }} {{/* Emoji to react with if wrong number AND Second Chance is true/on; if custom, use format name:id */}}
+{{ $IncorrectEmoji := "❌" }} {{/* Emoji to react with if number is incorrect; if custom, use format name:id */}}
+{{/* End of configuration values */}}
+
+{{/* No Touchy */}}
+{{ with .ExecData }}
+	{{ if not (getMessage (toInt (dbGet 0 "Channel").Value) .) }} {{/* Check if number was deleted */}}
+		{{ sendMessage (toInt (dbGet 0 "Channel").Value) (cembed 
+			"description" (print "<@" (dbGet 0 "LastUser").Value "> deleted their number which was correct!\nThe next number is " (dbGet 0 "Next").Value)
+			"color" 30654) }}
+	{{ return }}{{ end }}
+{{ end }}
+
+{{/* Initializing variables */}}
+{{ $LastUser := (dbGet 0 "LastUser").Value }}{{ $Next := (dbGet 0 "Next").Value }}
+{{ $Number := "" }}{{ $Number = round (slice ($Number = (exec "calc" (index .Args 0))) 9 (len $Number | add -1)) }}
+{{ $Wrong := "" }}{{ if $SecondChance }}{{ $Wrong = (dbGet 0 "CSecondChance").Value }}{{ end }}{{ $Silence := "" }}
+
+{{ if and (hasRoleID $Admin) (eq (toInt $Number) 0) (lt (toInt (dbGet 0 "Admin").Value) 1) }}
+	{{ dbSet 0 "Channel" (toString .Channel.ID) }}{{ dbSet 0 "LastUser" (toString .User.ID) }}{{ dbSet 0 "Next1" 1 }}{{ dbIncr 0 "Admin" 1 }}{{ addReactions $CorrectEmoji }}
+{{ return }}{{ end }}
+
+{{ if or (ne (toString $LastUser) (toString .User.ID)) $CountTwice }} {{/* Checks user */}}
+	{{ if eq (toInt $Next) (toInt $Number) }} {{/* Checks if correct number */}}
+		{{ addReactions $CorrectEmoji }}
+		{{ dbSet 0 "Next" (add $Next 1) }}
+		{{ dbSet 0 "LastUser" (toString .User.ID) }}
+		{{ if not $RoleID }} {{/* Counting role */}}
+			{{ takeRoleID $LastUser $RoleID }}
+			{{ giveRoleID .User.ID $RoleID }}
+		{{ end }}
+		{{ if $Leaderboard }} {{/* Updates leaderboard values */}}
+			{{ $Silence = dbIncr .User.ID "CCorrect" 1 }}
+			{{ $Silence = dbIncr .User.ID "CCount" 1 }}
+			{{ if gt (toInt $Number) (toInt (dbGet 0 "CHighScore").Value) }}
+				{{ dbSet 0 "CHighScore" (toInt $Number) }}
+				{{ dbSet 0 "CHSUser" (toString .User.ID) }}
+			{{ end }}
+		{{ end }}
+		{{ execCC .CCID nil 10 .Message.ID }} {{/* Check if number was deleted */}}
+		
+	{{ else }} {{/* Wrong number */}}
+		{{ if $SecondChance }}
+			{{ $Wrong = dbIncr 0 "CSecondChance" 1 }}
+			{{ if eq (toInt $Wrong) 1 }} {{/* Saved count */}}
+				{{ addReactions $WarningEmoji }}
+				{{ sendMessage nil (cembed 
+					"description" (print "<@" .User.ID "> sent an incorrect number of " $Number "\n**But**, second chance saved the count!\nThe next number is " $Next)
```

Comment:
The documentation says to run a setup command which initialises all needed database entries:

https://github.com/yagpdb-cc/yagpdb-cc/blob/c5662bc24501c2d7d10268a622422a272e771c93/website/docs/fun/counting/overview.md?plain=1#L50

The initialisation skips all this code, as the relevant branch is found here:

https://github.com/yagpdb-cc/yagpdb-cc/blob/c5662bc24501c2d7d10268a622422a272e771c93/src/fun/counting/main.go.tmpl#L33-L37

Reply:
Issue I see is before the admin command is run, users can still count, and be responded to. Users are told the next number is <nil>. If the admin command should be used in setup, maybe the command shouldn't run for anyone else until it's been initialized?

Reply:
While in theory this sounds like a good idea to implement, users adding custom commands from this repository are expected to read the whole documentation relevant to said custom command.

It could be made clearer by adding some sort of warning/caution around the relevant part, but I don't think it's worth it to change it around.

Reply:
The non-listed *option* to "just start counting" will initialize after one rotation of the system, counting incorrectly until reset. By doing this, the "Counting" database entry does not exist; this may cause issues with the `execCC` call as `channelID` would be nil.